pkg/fieldpath/paved.go

```diff
@@ -471,3 +471,113 @@ func (p *Paved) SetBool(path string, value bool) error {
 func (p *Paved) SetNumber(path string, value float64) error {
 	return p.SetValue(path, value)
 }
+
+// DeleteField deletes the field from the object.
+// If the path points to an entry in an array, the element
+// on that index is removed and the next ones are pulled
+// back. If it is a field on a map, the field is
+// removed from the map.
+func (p *Paved) DeleteField(path string) error {
```

Comment:
BTW, I could be convinced to rename this to `DeleteKey`.

Reply:
`Key` wouldn't be a good fit if array. What about `DeleteItem` ?

Reply:
Would simply `Delete` work? As in `paved.Delete` to delete whatever is at the supplied field path?

Reply:
@turkenh I feel like `Field` is more familiar than `Item` in this context for the users of fieldpath.

@negz I think it'd read nice if we had `Get`/`Set`/`Delete` but given that there is `GetValue`/`SetValue`, `paved.Delete` feels a bit off without a noun that says what you're deleting.

Thanks for the suggestions folks!